compiler/src/dotty/tools/dotc/core/tasty/LineSizesPickler.scala

```diff
@@ -0,0 +1,41 @@
+package dotty.tools
+package dotc
+package core
+package tasty
+
+import dotty.tools.tasty.TastyFormat.SOURCE
+import dotty.tools.tasty.TastyBuffer
+import TastyBuffer._
+
+import ast._
+import ast.Trees._
+import ast.Trees.WithLazyField
+import util.{SourceFile, NoSource}
+import core._
+import Contexts._, Symbols._, Annotations._, Decorators._
+import collection.mutable
+import util.Spans._
+
+class LineSizesPickler(pickler: TastyPickler) {
+
+  import ast.tpd._
+  val buf: TastyBuffer = new TastyBuffer(5000)
+  pickler.newSection("LineSizes", buf)
+
+  def pickleLineNumbers(source: SourceFile): Unit = {
+    val content = source.content()
+
+    val numLines = content.count(_ == '\n')
+    buf.writeInt(numLines + 1)
```

Comment:
I might be able to not pickle this one if we look at the size of the section when unpickling